src/test.zig

```diff
@@ -51,3 +51,116 @@ fn testDislayContents(test_dir_path: []const u8, archive_name: []const u8, file_
         try testing.expect(mem.eql(u8, archive.files.items[index].name, file_name));
     }
 }
+
+test "test suite" {
+    var ctx = TestContext.init(std.heap.page_allocator);
+    //simple test: create a new gnu archive and compare the output //doesn't work for now 
+    {
+       
+        
+        var case = try ctx.addCase("create gnu archive from txt files", "test1");
+        
+        try case.args.append("zar");
+        try case.args.append("-rc");
+        try case.args.append("./test.a");
+        try case.args.append("./input1.txt");
+        try case.testOutput();
+    }
+   
+
+    
+}
+pub const TestContext = struct {
+    //this struct manages the tests
+    const Self = @This();
+    cases: std.ArrayList(Case),
+    
+    pub const File = struct {
+            
+        fileType: FileType,
+        path: []const u8,
+        control: bool, 
+        
+        
+        pub const FileType = enum {
+            ar_bsd,
+            ar_gnu,
+            object, 
+            txt,
+        };
+    };
+    pub const Output = struct{
+        //struct 
+        stdout: ?[]const u8,
+        stderr: ?[]const u8,
+        outFile: File,
+        
+    };
+        
+    pub const Case = struct {
+        name: []const u8, 
+        input_files: std.ArrayList(File) = undefined,
+        expected_out: Output = undefined,
+        control: []const u8 = undefined,
+        args: std.ArrayList([]const u8) = std.ArrayList([]const u8).init(std.heap.page_allocator),
+        
+        pub fn init(name: []const u8) Case {
+            return .{ 
+                .name = name,
+            };
+        }
+        
+        pub fn deinit(self: Case) void {
+            self.args.deinit();
+        }
+            
+            
+        
+        pub fn testOutput(self: *Case) !void {
+            const result = try std.ChildProcess.exec(
+                .{
+                    .allocator = std.heap.page_allocator, 
+                    .argv = self.args.items,
+                    .cwd = "/home/jo/zar/test/data/test1",
```

Comment:
Hardcoded path, as already mentioned

Reply:
Yes, it is. I have put up this pr to show what I thought of and to have your go ahead to implement the rest of it.